File: math-skills-main/functions/variance.go
package functions

import(
	"strconv"
	"strings"
	"math"
)

func Variance(inputFile []byte) float64{
	mean := Average(inputFile)
	
	input := strings.Split(string(inputFile), "\n")
	inputStr := RemoveInvalid(input)

	var sum, count float64

	for i := 0; i < len(inputStr); i++{
		count++
		num, _ := strconv.ParseFloat(inputStr[i], 64) 
		diff := num - float64(mean)
		diff2 := (diff*diff)
		sum += diff2
	}
	result := sum/count
	output := math.Round(result)


	return output
}

File: helperfunctions/variance.c
// variance.c
#include "helpers.h"
#include <math.h>

double Calculate_variance(int data[], int n) {
    double mean = Calculate_mean(data, n);
    double sum = 0;
    for (int i = 0; i < n; ++i) {
        sum += pow(data[i] - mean, 2);
    }
    return sum / n;
}

